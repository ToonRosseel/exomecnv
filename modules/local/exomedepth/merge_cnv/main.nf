//MERGE CNV CALL FILES
process CNV_MERGE {
    tag "$meta"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/coreutils:9.3' :
        'biocontainers/coreutils:9.3' }"
    conda "${moduleDir}/environment.yml"

    publishDir "$params.outdir/exomedepth/cnv_call", mode: 'copy'

    input:
    tuple val(meta), path(auto), path(chrx)

    output:
    tuple val(meta), path("*.txt"), emit:merge
    path "versions.yml", emit:versions

    script:
    def prefix = task.ext.prefix ?: "${meta}_CNVs_ExomeDepth"
    """
    cp $auto "${prefix}.txt"
    tail +2 $chrx >> "${prefix}.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tail: \$(tail --version | sed '1!d; s/tail (GNU coreutils) //')
        cp: \$(cp --version | sed '1!d; s/cp (GNU coreutils) //')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta}_CNVs_ExomeDepth"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tail: \$(tail --version | sed '1!d; s/tail (GNU coreutils) //')
        cp: \$(cp --version | sed '1!d; s/cp (GNU coreutils) //')
    END_VERSIONS
    """
}
