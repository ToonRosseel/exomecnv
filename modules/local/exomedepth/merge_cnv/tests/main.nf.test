nextflow_process {

    name "Test Process CNV_MERGE"
    script "../main.nf"
    process "CNV_MERGE"

    test("homo_sapiens - bam, bai, roi") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.cnv_auto, checkIfExists:true),
                    file(params.cnv_chrx, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - bam, bai, roi - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.cnv_auto, checkIfExists:true),
                    file(params.cnv_chrx, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
