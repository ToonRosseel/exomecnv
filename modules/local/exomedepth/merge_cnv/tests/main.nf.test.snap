{
    "homo_sapiens - bam, bai, roi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "[id:Sample1, chr:autosomal, sam:[Sample1, Sample2, Sample3, Sample4, Sample5], fam:[Fam1, Fam2, Fam3, Fam4, Fam5]]_CNVs_ExomeDepth.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ce1a599ac7cf70d2997578484a4f4645"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "[id:Sample1, chr:autosomal, sam:[Sample1, Sample2, Sample3, Sample4, Sample5], fam:[Fam1, Fam2, Fam3, Fam4, Fam5]]_CNVs_ExomeDepth.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce1a599ac7cf70d2997578484a4f4645"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T13:04:50.037457512"
    },
    "homo_sapiens - bam, bai, roi - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Fam5]]_CNVs_ExomeDepth.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4425066ae908aabce62e256293297377"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Fam5]]_CNVs_ExomeDepth.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4425066ae908aabce62e256293297377"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T13:04:58.244422188"
    }
}